Druid-Tests/DRProductionBytecodeTest.class.st

```diff
@@ -1436,3 +1435,141 @@ DRProductionBytecodeTest >> testReturnTopFromMethod [
 	self executePrimitiveWithReceiver: object.
 	self assert: machineSimulator receiverRegisterValue equals: 17
 ]
+
+{ #category : #tests }
+DRProductionBytecodeTest >> testStoreAndPopReceiverVariableBytecode0 [
+
+	self testStoreAndPopReceiverVariableBytecode: 0
+]
+
+{ #category : #tests }
+DRProductionBytecodeTest >> testStoreAndPopReceiverVariableBytecode7 [
+
+	self testStoreAndPopReceiverVariableBytecode: 7
+]
+
+{ #category : #tests }
+DRProductionBytecodeTest >> testStoreAndPopReceiverVariableBytecode: index [
+
+	| object storeCheckTrampoline |
+	storeCheckTrampoline := self compile: [ cogit RetN: 0 ].
+	cogit ceStoreCheckTrampoline: storeCheckTrampoline.
+
+	self
+		compileBytecode: 200 + index
+		selector: #storeAndPopReceiverVariableBytecode
+		thenDo: [ :generator |
+			cogit ssPushRegister: TempReg.
+
+			"Execute the druid's compiled code"
+			generator value.
+
+			"Then return without druid's compiled code"
+			cogit genUpArrowReturn ].
+
+	object := self newObjectWithSlots: index + 1.
+
+	machineSimulator temporaryRegisterValue: memory trueObject.
```

Comment:
This goes to the remembered set in our case right?

Reply:
Nop, the object is in the new space, nothing to remember.